### PR TITLE
[OCPCLOUD-1859]: Update k8s packages to 1.26

### PR DESCRIPTION
**Pull request by odvarkadaniel**
To align with the 4.13 release, k8s dependencies need to be updated to 1.26

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,44 +10,35 @@ require (
...
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.25.2
-	k8s.io/apiextensions-apiserver v0.25.1
-	k8s.io/apimachinery v0.25.2
-	k8s.io/client-go v0.25.2
+	k8s.io/api v0.26.1
+	k8s.io/apiextensions-apiserver v0.26.1
+	k8s.io/apimachinery v0.26.1
+	k8s.io/client-go v0.26.1
 	k8s.io/cloud-provider-vsphere v1.25.0
```
Comment (JoelSpeed) [inline go.mod:23]: Should we update the cloud provider too?

Comment (odvarkadaniel) [reply go.mod:23]: This is the latest version of the package: https://pkg.go.dev/k8s.io/cloud-provider-vsphere

**Pull request merged**